node/network/bitfield-distribution/src/lib.rs

```diff
@@ -532,6 +542,9 @@ where
 			// get rid of superfluous data
 			state.peer_views.remove(&peerid);
 		}
+		NetworkBridgeEvent::NewGossipTopology(peers) => {
```

Comment:
Shouldn't this per relay chain head? So at session boundaries, we would gossip to different peers depending on the current head? I think that would be good for smooth session changes.

Reply:
Are you saying that during the session change we should gossip to two gossip group (from previous and current topology)?

Reply:
No, usually we gossip per head - right? So if we are sending messages with regard to head x, we should be using gossip peers that belong to the session of head x. Otherwise we lose the guarantee that we are going to reach everybody in two hops. I mean in practice it will  be fine, just because of the huge amount of redundancy.

Reply:
* newly added gossip peers should receive their updates as in 12de0155224bd3e9458bc60cf82f568aa7af27f0
* peers should receive `NewGossipTopology` on the first block that has new session index for child, which should happen roughly at the same time for peers. But if not, as you mentioned, redundancy helps.